code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/EventUtils.java

```diff
@@ -161,12 +161,12 @@ static boolean isAdIDEvent(final Event event) {
 	 * {@link IdentityConstants.Default#ZERO_ADVERTISING_ID} into {@code ""}.
 	 *
 	 * This method should not be used to detect whether the event is an ad ID event or not;
-	 * use {@link #isAdIDEvent(Event)} instead.
+	 * use {@link #isAdIdEvent(Event)} instead.
 	 *
 	 * @param event the event containing the advertising ID
 	 * @return the adID
 	 */
-	static String getAdID(final Event event) {
+	static String getAdId(final Event event) {
 		final Map<String, Object> data = event.getEventData();
 		try {
 			final String adID = (String) data.get(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER);
```

Comment:
Wrap this statement in try/catch for ClassCastException.

Reply:
Logic updated so that only adID variable is inside try/catch block